### PR TITLE
experiment: extended .most files

### PR DESCRIPTION
**Pull request by crusso**
Builds on #5879

Adds notion of multi-migration signature recording sequence of named migration function signatures followed by actor signature.

Eg.
``` motoko
// Version: 4.0.0
{
  "1" : {} -> {a : Nat};
  "2" : {} -> {b : Int};
  "3" : (old : {b : Int}) -> {b : Bool};
  "4" : (_old : {a : Nat}) -> {};
  "5" : (_old : {}) -> {a : Text}
}
actor  {
  stable a : Text;
  stable var b : Bool
} ;
```

* migration signatures ordered by lexicographic ordering of (text literal) labels "1", ... "n" (typically filenames)
* implement multi sig parsing, checking and printing.
* extend stable-compatibility relation to check post signature of old .most is compatible with appropriate pre signature of new .most.
* When comparing two signatures, we use the (last) migration function name of the old signature (if any) to determine the requirements of the next migration function in the new signature.
* reflexive on multi-migrations (mirroring idempotence)

TODO: 
[x] more simple tests (include negative tests for cmp/files/ehanced-migration*/)
[x] proper escaping of labels (done in another PR)

### CODE REVIEW

File: src/mo_frontend/typing.ml [hunk 1]
```diff
@@ -3995,7 +3996,7 @@ and validate_enhanced_migration_chain env stab_tfs at =
       if acc <> [] then begin
         let acc_fields = List.sort T.compare_field (List.map snd acc) in
         let acc_typ = T.Obj(T.Object, acc_fields, []) in
-        if not (T.sub acc_typ (T.normalize input)) then
+        if not (T.sub acc_typ (T.normalize input)) then (* CRUSSO: THIS NEED FIXING *)
```
Comment (alexandru-uta) [inline src/mo_frontend/typing.ml]: this a leftover?

Comment (crusso) [reply src/mo_frontend/typing.ml]: It's fixed higher up the stack

**Comment by crusso** [comment]
@alexandru-uta I forgot to commit some test files and have added them https://github.com/caffeinelabs/motoko/pull/5883/commits/1014a5e026f0c1a77d259c2b472a3998761a4a98
